src/Credentials.js

```diff
@@ -145,21 +147,31 @@ class Credentials {
   *  Send a push notification to a user, consumes a token which allows you to send push notifications
   *  and a url/uri request you want to send to the user.
   *
-  *  @param    {String}                  token       a push notification token (get a pn token by requesting push permissions in a request)
+  *  @param    {String}                  token              a push notification token (get a pn token by requesting push permissions in a request)
+  *  @param    {String}                  pubEncKey          the public encryption key of the receiver, encoded as a base64 string
+  *  @param    {String}                  payload.url        a uport request url
+  *  @param    {String}                  payload.message    a message to display to the user
   *  @return   {Promise<Object, Error>}              a promise which resolves with successful status or rejects with an error
   */
-  push (token, {url}) {
+  push (token, pubEncKey, {url, message}) {
```

Comment:
since function sig changes and push is still in uport-connect we will have update uport-connect at the same time, can be implemented in the background without changing func sig in uport-connect

Reply:
Yeah, I'm a little torn here. We could also put the `pubEncKey` as the last argument, and if it's not provided we fetch it from the registry. But I'm not sure if it's worth it since we will always have the pubEncKey from requestCredentials in uport-connect anyway.

Reply:
@oed not sure if it matters, if you don't have the pubEncKey here, you also don't have enough info to fetch from the registry either. Also in connect, if you know how to look up the registry, the object will already have the pubEncKey anyways, so need for the additional option.

I would maybe put the pubEncKey as the last arg, just to make it optional and backwards compatible with the current lib

Also why is there message and url now, not just url? (maybe saw this before, but forgot now)

Reply:
@zachferland I think the address is in the push token, so you could do a registry lookup.
I don't think we want to make it backwards compatible i.e. being able to send unencrypted push messages. So I don't really see a need to change the order.

The old pututu api supported url and message. The message would be displayed in the push notification. I'm hoping I can make it have the same behaviour with the new api.

Reply:
@oed ah okay, maybe, didn't know what was in the push token, if that is the case, then that might make sense. 

cool, so the url is basically the payload, and message is the text you see describing the push notification?

Reply:
@zachferland I think we should keep it as it is now. One tangential thing I want to do though is to introduce a session object in uport-connect which includes {address, pushToken, and PublicEncKey} this object can be retrived after a connection and provided on startup to remove the need to do a `requestCredentials` each time the page is reloaded.

yep, works as you said.

Reply:
yeah sounds good